autoload/coverage/lcov/parsing.vim

```diff
@@ -0,0 +1,130 @@
+" Copyright 2021 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+"{{{ Init
+
+let s:plugin = maktaba#plugin#Get('coverage')
+
+"}}}
+
+"{{{ [lcov](https://github.com/linux-test-project/lcov) coverage provider.
+
+""
+" @private
+" Categorizes a line potentially containing coverage information in a coverage
+" file.  Recognized coverage information is be formatted as:
+" - DA:<line number>,<execution count>[,<checksum>]
+" - BA:<line number>,<branch coverage (0: uncovered, 1: partial, 2: covered>
+" - BRDA:<line number>,<block number>,<branch number>,<taken>
+"
+" For BRDA, we can't get partial coverage (only '-' for uncovered, or 1+)
+function! s:TryParseLine(line) abort
+  if maktaba#string#StartsWith(a:line, 'BA:') ||
+        \ maktaba#string#StartsWith(a:line, 'DA:')
+    let l:hits_index = 1
+  elseif maktaba#string#StartsWith(a:line, 'BRDA:')
+    let l:hits_index = 3
+  else
+    return []
+  endif
+
+  try
+    let [l:prefix, l:raw_info] = split(a:line, ':')
```

Comment:
Prefer using split's `keepempty` option explicitly for more predictable behavior:
```vim
split(a:line, ':', 1)
```